src/main/java/com/gojek/beast/com/gojek/beast/protomapping/ProtoUpdateListener.java

```diff
@@ -0,0 +1,96 @@
+package com.gojek.beast.com.gojek.beast.protomapping;
+
+import com.gojek.beast.Clock;
+import com.gojek.beast.config.AppConfig;
+import com.gojek.beast.config.ColumnMapping;
+import com.gojek.beast.config.ProtoMappingConfig;
+import com.gojek.beast.converter.ConsumerRecordConverter;
+import com.gojek.beast.converter.RowMapper;
+import com.gojek.beast.models.ExternalCallException;
+import com.gojek.beast.models.UpdateBQTableRequest;
+import com.gojek.de.stencil.client.StencilClient;
+import com.gojek.de.stencil.parser.ProtoParser;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+@Slf4j
+public class ProtoUpdateListener extends com.gojek.de.stencil.cache.ProtoUpdateListener {
+    private ConsumerRecordConverter recordConverter;
+    private ProtoMappingConfig protoMappingConfig;
+    private AppConfig appConfig;
+    private String proto;
+    private StencilClient stencilClient;
+
+    public ProtoUpdateListener(ProtoMappingConfig protoMappingConfig, AppConfig appConfig, StencilClient stencilClient, String proto) throws ExternalCallException {
+        super(proto);
+        this.proto = proto;
+        this.protoMappingConfig = protoMappingConfig;
+        this.appConfig = appConfig;
+        this.stencilClient = stencilClient;
+        this.setProtoParser(getProtoMapping());
+    }
+
+    @Override
+    public void onProtoUpdate() {
+        log.info("updating bq table as {} proto was updated", this.getProto());
+        try {
+            this.updateProtoParser();
+        } catch (ExternalCallException e) {
```

Comment:
unless there's a conflict, we could call the methods directly, we can remove `this`.

On side note, we should make a list of conventions to follow.